### PR TITLE
Ajax Spider - add automation framework support

### PR DESCRIPTION
**Pull request by psiinon**
Signed-off-by: Simon Bennetts <psiinon@gmail.com>

### CODE REVIEW

File: addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJob.java [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+        AjaxSpiderTarget target = targetBuilder.build();
+		JobSpiderListener listener = new JobSpiderListener();
+
+		getExtSpider().startScan(uri.toString(), target, listener);
+
+        long endTime = Long.MAX_VALUE;
+        if (maxDuration > 0) {
+            // The spider should stop, if it doesnt we will stop it
+            endTime = System.currentTimeMillis() + (maxDuration * 1000 * 10) + 20;
```
Comment (kingthorin) [inline addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJob.java:156]: Use TimeUnit to make this a bit more readable?

File: addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJob.java [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+            } catch (InterruptedException e) {
+                // Ignore
+            }
+            if (!getExtSpider().isSpiderRunning()) {
+                break;
+            }
+            if (System.currentTimeMillis() > endTime) {
+                // It should have stopped but didn't (happens occasionally)
+            	getExtSpider().stopScan();
+                break;
```
Comment (kingthorin) [inline addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJob.java]: Something weird with the formatting (spacing) here.

File: addOns/spiderAjax/src/main/resources/org/zaproxy/zap/extension/spiderAjax/resources/Messages.properties [hunk 3]
```diff
@@ -143,3 +143,6 @@ spiderajax.toolbar.button.start     = New Scan
 spiderajax.toolbar.button.stop      = Stop AJAX Spider
 spiderajax.outofscope.response      = (403 Forbidden) Out of AJAX Spider scope
 spiderajax.warn.message.phantomjs.bug.invalid.target = The spider will not run.\nA bug in PhantomJS prevents the requests to localhost,\n127.0.0.1 and ::1 from being proxied through ZAP.
+
+spiderajax.automation.desc = Ajax Spider Automation Framework integration
```
Comment (kingthorin) [inline addOns/spiderAjax/src/main/resources/org/zaproxy/zap/extension/spiderAjax/resources/Messages.properties:147]:
```suggestion
spiderajax.automation.desc = Ajax Spider Automation Framework Integration
```

File: addOns/spiderAjax/src/test/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJobUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+
+        Control.initSingletonForTesting(Model.getSingleton(), extensionLoader);
+        Model.getSingleton().getOptionsParam().load(new ZapXmlConfiguration());
+    }
+
+    @Test
+    public void shouldReturnDefaultFields() {
+        // Given
+
+        // When
```
Comment (kingthorin) [inline addOns/spiderAjax/src/test/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJobUnitTest.java:102]:
```suggestion
        // Given / When
```
Combine these?

File: addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJob.java (hunk 1 above)
Comment (psiinon) [reply addOns/spiderAjax/src/main/java/org/zaproxy/zap/extension/spiderAjax/automation/AjaxSpiderJob.java:156]: Arg, should have remembered that - copied the spider code before making that fix ;)

**Comment by psiinon** [comment]
I need to fix the build ...

**Comment by psiinon** [comment]
I've disabled the tests. They work locally in eclipse but not from the command line or in the build route :(

**Comment by psiinon** [comment]
Arg, was still overriding the job name. Have fixed that now.
Should be good to review again...

**Comment by psiinon** [comment]
And just fixed a problem where the templates were not being loaded as the code was using the wrong resource location

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]

**Pull request merged**